### ISSUE TITLE
Crash when receiving smooch notifications with Smooch Chat window open.

### ISSUE THREAD

**Issue by andrewloyola** [smooch/smooch-ios#122]
scenario: in a flow hitting a button in chat triggers additional messages, we're seeing a consistent crash for one of our internal users when the new messages come in.  Other users seem to be unaffected.  this is with v5.4.0.  

not much in the stacktrace, other than the below:

`Jan 10 10:04:44 Maxwells-iPhone Openfolio(CoreFoundation)[3244] <Notice>: *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[SKTConversationActivity actions]: unrecognized selector sent to instance 0x170865b40'`

**Comment by wmora** [comment]
thanks for the info. Did you start getting this crash after updating to v5.4.0? Which version were you using before?

**Comment by andrewloyola** [comment]
yes as far as I can tell, it started in 5.4.0.  we went from 5.0.1 to 5.4.0

**Comment by wmora** [comment]
ok, and these additional messages that are triggered, do they have actions in them? Are they only text messages? Or a mix of both?

**Comment by genadyo** [comment]
Any idea if those 2 related? Started when I updated to 5.4.0 from 5.3.0.

http://crashes.to/s/012e7c450d6
http://crashes.to/s/7e5c1e5f635

**Comment by wmora** [comment]
@genadyo have you been able to reproduce it? Is it a similar scenario as the one reported by @andrewloyola ?

**Comment by andrewloyola** [comment]
@wmora the crashing messages are text only (technically a series 2-3 text messages then one message with 3 buttons, but as far as i can tell the initial text ones cause the issue)

**Comment by wmora** [comment]
hmmm can't reproduce it :-/

@andrewloyola I noticed in the stacktrace that the error is thrown by `[SKTConversationActivity actions]`. Are you sending any typing activity in between the messages? Are the messages being sent with a delay?

**Comment by andrewloyola** [comment]
yes we generally do send typing activity.  to clarify my original report, it's only happening (albeit consistently) for one of our users.   The rest on various devices but with the same app/smooch version and exact same message flows, seem to not exhibit the issue.

**Comment by genadyo** [comment]
It just happened to few beta testers and I rolled out to the previous version for the next release.
I'll wait to see more.

**Comment by wmora** [comment]
I was able to reproduce the crashes reported by @genadyo and just released version [5.4.2](https://github.com/smooch/smooch-ios/releases/tag/5.4.2) with the fix. 

As for the original issue reported by @andrewloyola, I'm still looking into it. What is the device model and iOS version running on the device with the crash?

**Comment by wmora** [comment]
I'm closing this issue for now. @andrewloyola feel free to reopen it if you still get the crash with the latest version of the SDK

**Comment by andrewloyola** [comment]
hi @wmora sorry for the radio silence on this.  I was able to grab the user's phone to re-test with the latest smooch SDK (5.5.2) recently and i see the same thing.  I was able to grab a better stacktrace this time though.  It  looks like a crash in fetchCreditCardInfoIfNecessary.  (see below):

```
(lldb) bt
* thread #1: tid = 0x7059b, 0x000000018f800524 libobjc.A.dylib`objc_exception_throw, stop reason = breakpoint 1.2
    frame #0: 0x000000018f800524 libobjc.A.dylib`objc_exception_throw
    frame #1: 0x0000000190dd0268 CoreFoundation`-[NSObject(NSObject) doesNotRecognizeSelector:] + 140
    frame #2: 0x0000000190dcd270 CoreFoundation`___forwarding___ + 916
    frame #3: 0x0000000190cc680c CoreFoundation`_CF_forwarding_prep_0 + 92
    frame #4: 0x00000001017f06d4 Smooch`-[SKTConversationViewController fetchCreditCardInfoIfNecessary:] + 196
    frame #5: 0x00000001017f04ac Smooch`-[SKTConversationViewController configureMessageCell:forMessageAtIndex:] + 664
  * frame #6: 0x00000001017ae2c4 Smooch`-[SKTSOMessagingViewController tableView:cellForRowAtIndexPath:] + 1472
    frame #7: 0x000000010026b0bc OFDevS`_logos_method$TVDataSourceModCore$__ABT$tableView$cellForRowAtIndexPath$(self=<unavailable>, _cmd="tableView:cellForRowAtIndexPath:", tableView=0x000000010395ae00, indexPath=0xc000000005c00116) + 236 at ABTWYSIWYGViewControllerSwizzling.xm:433
    frame #8: 0x0000000196fc1b58 UIKit`-[UITableView _createPreparedCellForGlobalRow:withIndexPath:willDisplay:] + 716
    frame #9: 0x0000000196fc1d88 UIKit`-[UITableView _createPreparedCellForGlobalRow:willDisplay:] + 80
    frame #10: 0x0000000196faf320 UIKit`-[UITableView _updateVisibleCellsNow:isRecursive:] + 2304
    frame #11: 0x0000000196fc6dec UIKit`-[UITableView _performWithCachedTraitCollection:] + 116
    frame #12: 0x0000000196d62de8 UIKit`-[UITableView layoutSubviews] + 176
    frame #13: 0x0000000196c7ba80 UIKit`-[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1196
    frame #14: 0x00000001941299d8 QuartzCore`-[CALayer layoutSublayers] + 148
    frame #15: 0x000000019411e4cc QuartzCore`CA::Layer::layout_if_needed(CA::Transaction*) + 292
    frame #16: 0x000000019411e38c QuartzCore`CA::Layer::layout_and_display_if_needed(CA::Transaction*) + 32
    frame #17: 0x000000019409b3e0 QuartzCore`CA::Context::commit_transaction(CA::Transaction*) + 252
    frame #18: 0x00000001940c2a68 QuartzCore`CA::Transaction::commit() + 512
    frame #19: 0x00000001940c3488 QuartzCore`CA::Transaction::observer_callback(__CFRunLoopObserver*, unsigned long, void*) + 120
    frame #20: 0x0000000190d760c0 CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 32
    frame #21: 0x0000000190d73cf0 CoreFoundation`__CFRunLoopDoObservers + 372
    frame #22: 0x0000000190d74180 CoreFoundation`__CFRunLoopRun + 1024
    frame #23: 0x0000000190ca22b8 CoreFoundation`CFRunLoopRunSpecific + 444
    frame #24: 0x0000000192756198 GraphicsServices`GSEventRunModal + 180
    frame #25: 0x0000000196ce97fc UIKit`-[UIApplication _run] + 684
    frame #26: 0x0000000196ce4534 UIKit`UIApplicationMain + 208
    frame #27: 0x00000001001a4e78 OFDevS`main + 144 at AppDelegate.swift:64
    frame #28: 0x000000018fc855b8 libdyld.dylib`start + 4

```

**Comment by wmora** [comment]
thanks @andrewloyola, I'll look into it as soon as I can

**Comment by wmora** [comment]
should be fixed with version [5.6.7](https://github.com/smooch/smooch-ios/releases/tag/5.6.7)
